lib/biokbase/catalog/db.py

```diff
@@ -1301,34 +1416,37 @@ def check_db_schema(self):
                 'Incompatible DB versions.  Expecting DB V4, found DV V' + str(db_version) +
                 '. You are probably running an old version of the service.  Start up failed.')
 
-    def get_db_version(self):
-        # version is a collection that should only have a single 
-        version_collection = self.db[MongoCatalogDBI._DB_VERSION]
+    def get_db_version(self, db):
+
+        # version is a collection that should only have a single
+        version_collection = db[MongoCatalogDBI._DB_VERSION]
         ver = version_collection.find_one({})
         if (ver):
             return ver['version']
         else:
             # if there is no version document, then we are DB v1
-            self.update_db_version(1)
+            self.update_db_version(1, db)
             return 1
 
-    def update_db_version(self, version):
+    def update_db_version(self, version, db):
+
         # make sure we can't have two version documents
-        version_collection = self.db[MongoCatalogDBI._DB_VERSION]
+        version_collection = db[MongoCatalogDBI._DB_VERSION]
         version_collection.create_index('version_doc', unique=True, sparse=False)
         version_collection.update_one({'version_doc': True}, {'$set': {'version': version}},
                                       upsert=True)
 
     # version 1 kept released module versions in a map, version 2 updates that to a list
     # and adds dynamic service tags
-    def update_db_1_to_2(self):
-        for m in self.modules.find({'release_versions': {'$exists': True}}):
+    def update_db_1_to_2(self, db):
+        modules_collection = db[MongoCatalogDBI._MODULES]
+        for m in modules_collection.find({'release_versions': {'$exists': True}}):
             release_version_list = []
```

Comment:
Why can't we delete all these upgrade methods again?

Reply:
https://kbase.slack.com/archives/C05SSJT13BN/p1739471119225879?thread_ts=1739292839.872669&cid=C05SSJT13BN

https://kbase.slack.com/archives/C05SSJT13BN/p1739471521100149?thread_ts=1739292839.872669&cid=C05SSJT13BN

Reply:
Ok, so the answer is that when you remove them tests start failing. Can you make an issue in that case?

Reply:
https://github.com/kbase/catalog/pull/140#discussion_r1955046184

Reply:
Oh great, forgot about that. :+1: